packages/metadata-common/src/MetadataEditor.tsx

```diff
@@ -123,7 +123,8 @@ export class MetadataEditor extends ReactWidget {
     for (const schemaField in this.schema) {
       if (
         this.requiredFields.includes(schemaField) &&
-        (this.metadata[schemaField] === null ||
+        (this.metadata[schemaField] === undefined ||
+          this.metadata[schemaField] === null ||
           this.metadata[schemaField] === '' ||
```

Comment:
What if you do something like  `&& (this.metadata[schemaField] || this.metadata[schemaField] === '(No selection)')` where the first comparison should take care of null, undefined and empty string

Reply:
How about a local function `this.is_field_set(schemaField: str): boolean`?  Seems like there are way too many variations to deal with.

Reply:
@lresende we can't do that because 0 or false may be valid and are also considered falsy when evaluating `this.metadata[schemaField]`, also `[]` is considered truthy. I do like the idea of a separate function like @kevin-bates suggested, I'll do that.

Reply:
pushed an update